### PR TITLE
ARROW-5818: [Java][Gandiva] support varlen output vectors

### PR DESCRIPTION
**Pull request by pravindra**
callback to java for resizing varlen vectors

### CODE REVIEW

**Comment by pravindra** [comment]
@praveenbingo @pprudhvi @shyambits2004, please review.

File: cpp/src/gandiva/jni/jni_common.cc [hunk 1]
```diff
@@ -755,13 +816,15 @@ Java_org_apache_arrow_gandiva_evaluator_JniWrapper_evaluateProjector(
       uint8_t* value_buf = reinterpret_cast<uint8_t*>(out_bufs[buf_idx++]);
       jlong data_sz = out_sizes[sz_idx++];
       if (arrow::is_binary_like(field->type()->id())) {
-        buffers.push_back(std::make_shared<JavaResizableBuffer>(value_buf, data_sz));
+        buffers.push_back(std::make_shared<JavaResizableBuffer>(
```
Comment (praveenbingo) [inline cpp/src/gandiva/jni/jni_common.cc]: should we add a null check for expander in case of varlength output..

**Review by praveenbingo — APPROVED** [review]
Looks good otherwise.

File: cpp/src/gandiva/jni/jni_common.cc (hunk 1 above)
Comment (pravindra) [reply cpp/src/gandiva/jni/jni_common.cc]: done

**Pull request closed without merging**